modules/widgets/src/compass-widget.tsx

```diff
@@ -45,10 +45,13 @@ export class CompassWidget implements Widget<CompassWidgetProps> {
     this.id = props.id || 'compass';
     this.viewId = props.viewId || null;
     this.placement = props.placement || 'top-left';
-    props.transitionDuration = props.transitionDuration || 200;
-    props.label = props.label || 'Compass';
-    props.style = props.style || {};
-    this.props = props;
+
+    this.props = {
+      ...props,
+      transitionDuration: props.transitionDuration || 200,
+      label: props.label || 'Compass',
+      style: props.style || {}
+    };
```

Comment:
Construct new props instead of mutating them. It's a better practice, plus react throws an error.

Reply:
Moved to https://github.com/visgl/deck.gl/pull/9312